### ISSUE TITLE
DB flags --driver --host etc. ignored without configuration file

### ISSUE THREAD

**Issue by cben** [joereynolds/gauxilium#141]
https://github.com/joereynolds/sql-lint/blob/e9b373a57539e13c335129b4dfc4635c5d02f5f9/src/main.ts#L100-L114

These flags are only taken into account when merging them with values from a config file IF there is a config file.
I'd expect them to be alternatives, if I'm providing all flags on command line, I don't need config file?  This would be consistent with the fact that when both are present, CLI flags override the config.

In other words, the behavior I propose is: treat missing config file as empty config file; always merge flags || configuration; connect to a DB iff host is set.

cc @zvikorn @vkareh

**Comment by vkareh** [comment]
Good point. Seems like lines 102 and 103 should be removed:
```diff
 if (configuration === null) { 
   printer.warnAboutNoConfiguration(file); 
-   runner.run(queries, printer, prefix, omittedErrors); 
-   process.exit(0); 
 } 
```
That way it warns the user, but then uses the cli-then-config-then-defaults mode.

**Comment by joereynolds** [comment]
Great point @cben, that's slipped my mind as I've always ran with a config file in place. PRs are welcome (with tests ideally) and I'll add it to the [0.0.14 milestone](https://github.com/joereynolds/sql-lint/milestone/4) :+1:

**Comment by joereynolds** [comment]
@vkareh your solution will mostly work but if you have no configuration whatsoever and pass no flags then the connection will not be established and will dump out errors to the console.

**Comment by joereynolds** [comment]
Hey @cben and @vkareh,
This has been fixed, if you want to test it, I'd be grateful.

https://github.com/joereynolds/sql-lint/commit/ea1706d02f9e4bfa3e847dea56faa1b062ed4c75

Cheers!
